handler/src/main/java/io/netty/handler/ssl/OpenSslServerSessionContext.java

```diff
@@ -32,36 +31,48 @@ public void setSessionTimeout(int seconds) {
         if (seconds < 0) {
             throw new IllegalArgumentException();
         }
-        SSLContext.setSessionCacheTimeout(context.ctx, seconds);
+        synchronized (context) {
+            SSLContext.setSessionCacheTimeout(context.ctx, seconds);
```

Comment:
@normanmaurer - followup is fine in tcnative ... consider having the `free` methods (for context and ssl) return something we can assign to the long pointer in java (e.g. NULL) ... so we don't have to hard code `0` in java.

```c
TCN_IMPLEMENT_CALL(jint, SSLContext, free)(TCN_STDARGS, jlong ctx)
{
    tcn_ssl_ctxt_t *c = J2P(ctx, tcn_ssl_ctxt_t *);
    UNREFERENCED_STDARGS;
    TCN_ASSERT(ctx != 0);
    /* Run and destroy the cleanup callback */
    int result = apr_pool_cleanup_run(c->pool, c, ssl_context_cleanup);
    apr_pool_destroy(c->pool);

    // MODIFIED
    if (result != <expected value) {
       // throw exception?
    }
    return NULL;
    // MODIFIED
}
```

```java
ctx = SSLContext.free(ctx);
```